components/JoinGroupModal.js

```diff
@@ -31,75 +30,40 @@ function reducer(state, action) {
         ...state,
         success: action.value,
       };
-    case 'setFirstName':
-      return {
-        ...state,
-        firstName: action.value,
-      };
-    case 'setLastName':
-      return {
-        ...state,
-        lastName: action.value,
-      };
-    case 'setEmail':
-      return {
-        ...state,
-        email: action.value,
-      };
     default:
       throw new Error();
   }
 }
 
-export default (props) => {
-  const [
-    { loading, success, firstName, lastName, email },
-    dispatch,
-  ] = useReducer(reducer, initialState);
-  const dropdownAlertRef = useRef(null);
+export default function JoinGroupModal({ groupID, title, showSuccess }) {
+  const [{ loading, success }, dispatch] = useReducer(reducer, initialState);
 
   const handleOpenModal = () => {
     Amplitude.logEventWithProperties('OPEN Group Sign Up', {
-      group: props.title,
+      group: title,
     });
   };
 
-  const handleSignUp = () => {
-    if (!firstName || !lastName || !email) {
-      return dropdownAlertRef.current.alertWithType(
-        'warn',
-        '😮',
-        `Make sure you've filled out everything`
-      );
-    }
-
-    if (!isEmailValid(email)) {
-      return dropdownAlertRef.current.alertWithType(
-        'error',
-        '🤔',
-        'Make sure your email is correct'
-      );
-    }
-
+  const handleSignUp = ({ firstName, lastName, email }) => {
     dispatch({ type: 'setLoading', value: true });
     Amplitude.logEventWithProperties('SUBMIT Group Sign Up', {
-      group: props.title,
+      group: title,
     });
-    joinGroup(props.groupID, firstName, lastName, email)
+    joinGroup(groupID, firstName, lastName, email)
       .then((joinGroupSuccess) => {
         if (joinGroupSuccess) {
           // close this modal
           dispatch({ type: 'setLoading', value: false });
           dispatch({ type: 'setSuccess', value: true });
           setTimeout(() => dispatch({ type: 'setSuccess', value: false }), 0);
 
-          props.showSuccess('Thanks for joining ☺️');
+          showSuccess('Thanks for joining ☺️');
         }
       })
       .catch((err) => {
         dispatch({ type: 'setLoading', value: false });
         Amplitude.logEventWithProperties('ERROR Group Sign Up', {
-          group: props.title,
+          group: title,
           error: err,
         });
```

Comment:
Do you want to show some kind of error message to the user? So they know they didn't actually join the group?

Reply:
this should basically never happen 🤞 
but i can add some basic error handling in a separate PR